src/Controllers/DiagramController.php

```
@@ -219,47 +219,99 @@ public function generate(): void
-                if (isset($okrTitles[$nodeKey]) || isset($okrDescriptions[$nodeKey])) {
-                    DiagramNode::update($this->db, (int) $node['id'], [
-                        'okr_title'       => $okrTitles[$nodeKey] ?? '',
-                        'okr_description' => $okrDescriptions[$nodeKey] ?? '',
-                    ]);
+        $nodes = [];
+        $this->db->beginTransaction();
+        try {
+            // Update the diagram code (version is incremented automatically)
+            StrategyDiagram::update($this->db, (int) $diagram['id'], [
+                'mermaid_code' => $mermaidCode,
+            ]);
```

PHPMD (2.15.0)

[error] Avoid using static access to class '\StratFlow\Models\StrategyDiagram' in method 'save'. (undefined)

(StaticAccess)